components/director/internal/open_resource_discovery/validation.go

```diff
@@ -604,6 +626,42 @@ func validateEventInputWithSuppressedErrors(event *model.EventDefinitionInput, e
 		})))
 }
 
+func validateCapabilityInput(capability *model.CapabilityInput) error {
+	return validation.ValidateStruct(capability,
+		validation.Field(&capability.OrdPackageID, validation.Required, validation.Match(regexp.MustCompile(PackageOrdIDRegex))),
+		validation.Field(&capability.Name, validation.Required),
+		validation.Field(&capability.Description, validation.Required, validation.Length(MinDescriptionLength, MaxDescriptionLength)),
+		validation.Field(&capability.OrdID, validation.Required, validation.Match(regexp.MustCompile(CapabilityOrdIDRegex))),
+		validation.Field(&capability.Type, validation.Required, validation.In(CapabilityTypeCustom, CapabilityTypeMDICapabilityDefinitionV1), validation.When(capability.CustomType != nil, validation.In(CapabilityTypeCustom))),
```

Comment:
Add validation check when type is `sap.mdo:mdi-capability:v1` the capability definition type must be - `sap.mdo:mdi-capability-definition:v1`